### PR TITLE
Introduce themable colors for resolved and unresolved comments

### PR DESCRIPTION
**Pull request by hermannloose**
Implementation proposal for #127473.

### CODE REVIEW

**Review by alexr00 — CHANGES_REQUESTED** [review]
Looks and works great! I left a few comments, but I intend to push changes to address them today so that I can merge this before we start our endgame next week.

After our endgame week we'll do a UX review, try to get feedback from other extension authors, and work towards finalizing the API.

**Review by alexr00 — APPROVED** [review]
I made the changes I commented on so that we can merge this before our endgame starts. Thank you for the PR!

**Comment by alexr00** [comment]
There's an outage that's preventing me from merging this right now, but I will merge it when that's over.

File: src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts [hunk 1]
```diff
@@ -56,20 +57,8 @@ export const COMMENTEDITOR_DECORATION_KEY = 'commenteditordecoration';
...
-	[languages.CommentThreadState.Resolved, resolvedCommentBorder],
-]);
-
-const commentThreadStateColorVar = '--comment-thread-state-color';
-
-function getCommentThreadStateColor(thread: languages.CommentThread, theme: IColorTheme): Color | undefined {
-	const colorId = thread.state !== undefined ? commentThreadStateColors.get(thread.state) : undefined;
-	return colorId !== undefined ? theme.getColor(colorId) : undefined;
+function getCommentThreadWidgetStateColor(thread: languages.CommentThread, theme: IColorTheme): Color | undefined {
+	return getCommentThreadStateColor(thread, theme) ?? theme.getColor(peekViewBorder);
```
[the remark this answers is not in the document]
Comment (alexr00) [reply src/vs/workbench/contrib/comments/browser/commentThreadWidget.ts:61]: I split it out in preparation for including the color in the "Comments" view. There, we will not use the peek view color, but I do expect that we still use the same resolved/unresolved colors.

**Comment by hermannloose** [comment]
I just realized I had forgotten to upload the local changes I had made yesterday to base the unresolved colors on `editorWarningBackground`, sorry for causing the extra work. 😅

**Pull request merged**